client/state/ui/layout-focus/reducer.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_FOCUS_SET,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_NEXT_FOCUS_SET,
+} from 'state/action-types';
+import { setFocusHideClass } from 'lib/layout-focus';
+
+const initialState = { current: null, previous: null, next: null };
+const validAreas = [ 'content', 'sidebar', 'sites', 'preview' ];
+const isValidArea = area => includes( validAreas, area );
+
+export default function layoutFocus( state = initialState, action ) {
+	switch ( action.type ) {
+		case LAYOUT_FOCUS_SET:
+			if ( ! isValidArea( action.area ) ) {
+				return state;
+			}
+			setFocusHideClass();
```

Comment:
I'm trying to think of ways we can avoid this call being here. Could we create a Redux middleware for layoutFocus which watches for the `LAYOUT_FOCUS_SET` action?